### PR TITLE
[PLACES-232] Rebuild migration for database and Rework API to handle new Schema.

### PR DESCRIPTION
**Pull request by sublet**
Big update.

Integrated a new migration that will give us a good foundation for the API.  This touches pretty much everything, and I will go through and update the spec so that it is aligned.

Additional benefit from this was that we are now using mock data for the entire API, so no longer have to see that data.

### CODE REVIEW

File: expressjs/db/migrations/20200526095904_rebuild.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    DROP TYPE if exists state_type;
+    DROP TABLE if exists organizations;
+  `;
+  return knex.raw(dropQuery);
+}
+
+function buildOrganization(knex) {
+  return knex.schema.createTable('organizations', function (table) {
+    table.uuid('id').notNull().primary();
+    table.string('authority_name')
```
Comment (MarshallMcCoy) [inline expressjs/db/migrations/20200526095904_rebuild.js:18]: This should potentially be just `name`. A health authority is comprised of many organizations so the use `authority` here should be a little misleading.

File: expressjs/db/migrations/20200526095904_rebuild.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    table.uuid('organization_id').notNull().references('organizations.id').onDelete('CASCADE');
+    table.string('info_website_url')
+    table.string('reference_website_url')
+    table.string('api_endpoint_url')
+    table.json('region_coordinates')
+    table.integer('notification_threshold_percent').defaultTo(66);
+    table.integer('notification_threshold_count').defaultTo(6);
+    table.integer('chunking_in_seconds').defaultTo(43200);
+    table.timestamp('updated_at').defaultTo(knex.fn.now())
+    table.timestamp('created_at').defaultTo(knex.fn.now())
```
Comment (MarshallMcCoy) [inline expressjs/db/migrations/20200526095904_rebuild.js:36]: We should have a field on this table that represents the number of days to retain records. Probably just a column named `numberOfDaysToRetainRecords` that stores an `Int`.

File: expressjs/app/api/safepaths/controller.js [hunk 3]
```diff
@@ -15,35 +15,36 @@ const publicationFiles = require('../../lib/publicationFiles');
...
             'Content-Length': data.length
           })
           .send(data)
+      } else if (type === 'json') {
+        let response = publicationFiles.build(organization, publication, intervalTrails)
+        res.status(200).json(response);
       }
       
-      let response = publicationFiles.build(organization, record, intervalTrails)
+      // TODO: Save to disk.
```
Comment (krishnadurai) [inline expressjs/app/api/safepaths/controller.js:46]: What are we intending to save to disk in here?

File: expressjs/db/migrations/20200526095904_rebuild.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+function buildPublications(knex) {
+  return knex.schema.createTable('publications', function (table) {
+    table.increments('id').notNull().primary();
+    table.uuid('organization_id').notNull().references('organizations.id').onDelete('CASCADE');
+    table.timestamp('start_date').notNullable();
+    table.timestamp('end_date').notNullable();
+    table.timestamp('publish_date').notNullable();
+    table.timestamp('updated_at').defaultTo(knex.fn.now())
+    table.timestamp('created_at').defaultTo(knex.fn.now())
```
Comment (krishnadurai) [inline expressjs/db/migrations/20200526095904_rebuild.js:64]: nit: ';' missing

File: expressjs/db/migrations/20200526095904_rebuild.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+function buildTrails(knex) {
+  return knex.schema.createTable('trails', function (table) {
+    table.increments('id').notNull().primary();
+    table.uuid('case_id').notNull().references('cases.id').onDelete('CASCADE');
+    table.specificType('coordinates', 'geometry(point, 4326)');
+    table.timestamp('time');
+    table.string('hash')
+    table.timestamps(true, true);
```
Comment (krishnadurai) [inline expressjs/db/migrations/20200526095904_rebuild.js:96]: Should we change this to the same pattern?
```
    table.timestamp('updated_at').defaultTo(knex.fn.now());
    table.timestamp('created_at').defaultTo(knex.fn.now());
```

**Comment by jvisenti** [comment]
We also need to update `organizations_seed` to match the new schema. Should just mean dropping the `info_website` and `api_endpoint_url` fields in the seed.

File: expressjs/app/api/safepaths/controller.js (hunk 3 above)
Comment (sublet) [reply expressjs/app/api/safepaths/controller.js:46]: We will be.  That's next on my list.

File: expressjs/db/migrations/20200526095904_rebuild.js (hunk 5 above)
Comment (sublet) [reply expressjs/db/migrations/20200526095904_rebuild.js:96]: Yes, good catch.

**Comment by jvisenti** [comment]
Also looks like `test/cases.js` needs to be updated for the new schema (tests are failing)

**Comment by sublet** [comment]
Thanks @jvisenti .  Updated to handle.

**Pull request merged**